import Note from "../models/Notes.js"

//fetch all
export const getAllNotes = async(_, res) => {
  try {
    const notes = await Note.find().sort({createdAt:-1});
    res.status(200).json(notes);
  } catch (error) {
    console.error("Error in getAllNotes controller", error); 
    res.status(500).json({message:"internal server error"});
  }
}

// fetch by id
export const getNoteById = async(req, res) => {
try {
  const note = await Note. findById(req.params.id);
  if(!note) return res.status(404).json({message:"Note not found"});
  res.json(note);

  } catch (error) {
   console.error("Error in getNoteId controller", error); 
    res.status(500).json({message:"internal server error"});
  }
}

export const createNotes = async(req, res) => {
  try {
    
    const {title,content} = req.body
    const newNote = new Note({title, content})
     const saveNote = await newNote.save()
    
    
    res.status(201).json(saveNote)
  } catch (error) {
     console.error("Error in createNotes controller", error); 
    res.status(500).json({message:"internal server error"});
  }
}

export const updateNotes = async(req, res) => {
  try {
    const {title, content} = req.body
   const updateNote = await Note.findByIdAndUpdate(req.params.id,{title, content},{
    new:true
   }
  );

   if(!updateNote) return res.status(404).json({message:"Note not found"});

    res.status(200).json(updateNote);
  } catch (error) {
      console.error("Error in updateNotes controller", error); 
    res.status(500).json({message:"internal server error"});
  }
  
}

 export const deleteNotes = async(req, res) => {
  try {
    const deletedNote = await Note.findByIdAndDelete(req.params.id);

    if(!deleteNotes) return res.status(404).json({message: "Note not found"});

    res.status(200).json({message: "Note deleted successfully!"});
  } catch (error) {
   console.error("Error in deleteNotes controller", error); 
    res.status(500).json({message:"internal server error"});
  }
}